### PR TITLE
fix(Testing/SlimCheck): allow sampling from higher universes

### PR DESCRIPTION
**Pull request by eric-wieser**
This restores a Lean 3 feature by addressing a porting note, and allows `slim_check` to find a counterexample to
```lean
example (α : Type u) (l : List α) : l = l ++ l := by slim_check
```

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by YaelDillies — APPROVED** [review]
This looks fine to me, but I know that @kim-em had beef with it.

maintainer merge

**Comment by kim-em** [comment]
I think Henrik is still in the process of extracting SlimCheck out to a separate repo, so please don't merge without hearing from @hargoniX.

File: Mathlib/Testing/SlimCheck/Sampleable.lean [hunk 1]
```diff
@@ -275,15 +280,22 @@ instance sampleableExt [SampleableExt α] [Repr α] : SampleableExt (NoShrink α
...
 Print (at most) 10 samples of a given type to stdout for debugging.
 -/
--- Porting note: if `Control.ULiftable` is ported, make use of that here, as in mathlib3,
--- to enable sampling from higher types.
-def printSamples {t : Type} [Repr t] (g : Gen t) : IO PUnit := do
-  for i in List.range 10 do
-    IO.println s!"{repr (← g.run i)}"
+def printSamples {t : Type u} [Repr t] (g : Gen t) : IO PUnit :=
+  -- TODO: this should be a global instance
+  letI : MonadLift Id IO := ⟨fun f => pure <| Id.run f⟩
```
Comment (kmill) [inline Mathlib/Testing/SlimCheck/Sampleable.lean:288]: I discounted https://github.com/leanprover/lean4/pull/5638, but maybe there's a use for it after all?

Comment (eric-wieser) [reply Mathlib/Testing/SlimCheck/Sampleable.lean:288]: The use here arises when working with monad stacks, and wanting to lift from `StateT S (ReaderT R Id)` to `StateT S (ReaderT R IO)` or similar.

**Comment by Command-Master** [comment]
This should move to [Plausible](https://github.com/leanprover-community/plausible)

**Pull request closed without merging**